client/apps/edit/components/header/index.jsx

```diff
@@ -0,0 +1,187 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import Icon from '@artsy/reaction-force/dist/Components/Icon'
+import { data as sd } from 'sharify'
+
+export class EditHeader extends Component {
+  static propTypes = {
+    actions: PropTypes.any,
+    article: PropTypes.object,
+    channel: PropTypes.object,
+    edit: PropTypes.object,
+    user: PropTypes.object
+  }
+
+  toggleSection = (activeSection) => {
+    this.props.actions.changeSection(activeSection)
```

Comment:
This is probably a naive question due to lack of understanding, but but why pass in `props.actions` from parent to child rather than using `connect`?

Reply:
Just new to redux I think? Hadn't consider connecting components inside the outer container.
Moved this component into a separate PR but will check out here after it is merged.

Reply:
Oh yeah, absolutely nothing should be passed :) Thats the beauty of redux, you connect each sub component individually to the store and everything remains decoupled. With no props passing you can move stuff around at will and the component tree remains really clean with only things like 

```javascript 
const App = () => 
  <Provider store={store}>
    <Layout />
  </Provider>

// Layout.js

function Layout ({ page }) {
  return (
    <div>
      <div>
        Hello, you are on step {this.props.page}
      </div>

      <NextButton />
      <PreviousButton  />
    </div>
  )
}

const mapStateToProps = (state) => ({
  page: state.onboarding.page
})

export default connect(
  mapStateToProps
)(Layout) 

// NextButton.js 

function NextButton ({ nextButtonEnabled, nextButtonAction }) {
  return (
    <button enabled={nextButtonEnabled} onClick={nextPageAction}>
       Next
    </button>
  )
}

const mapStateToProps = (state) => ({
  nextButtonEnabled: state.onboarding.nextButtonEnabled
})

const mapDispatchToProps = {
  nextPageAction: nextPage
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NextButton) 

... 
```

As things scale out the benefit of co-locating data deps with components really shows and helps with refactors and maintainability since you can move components around at will.

Reply:
Note that the above form can be written any number of ways -- and if you use classes you can use decorators to keep things really minimal:

```javascript
@connect(state => {
  return {
    page: state.onboarding.page,
    totalPages: state.onboarding.totalPages,
    ...
  }
})
export class App extends Component {
  render() {
    const { page, totalPages } = this.props 

    return (
      <div>
        <div>
          Hello, you are on step {page} of {totalPages}
        </div>

        <NextButton />
        <PreviousButton  />
      </div>
    )
  }
}
```

Reply:
cc @kanaabe